### ISSUE TITLE
Master app not pulling in configs from apps under umbrella.

### ISSUE THREAD

**Issue by seantanly** [bitwalker/distillery-umbrella-test#5]
Just tried out the updated version and here's some observations.
1. [Need clarification] I believe master_app/config/config.exs is supposed to hold configurations that is meant to override the individual apps default configuration values. Thus, app_one and app_two should be given working defaults instead of nil for port. And master_app can override them to different values.
2. With this in mind, I removed the configs in master_app and restored the defaults in app_one and app_two for testing. Without running conform, executing the release based on master_app, doesn't pick up the port configs in app_one, app_two.
3. [Need fix] Running `conform.new` crashes when master_app/config/config.exs doesn't contain any configuration. Adding a single config allows `conform.new` to execute successfully.

```
$ mix conform.new
==> ranch (compile)
==> cowlib (compile)
==> cowboy (compile)
==> The schema for your project has been placed in config/app_one.schema.exs
==> The schema for your project has been placed in config/app_two.schema.exs
** (ArgumentError) argument error
    :erlang.byte_size(["[", "extends: ", "[]", ",", "\n ", "import: ", "[]", ",", "\n ", "mappings: ", "[]", ",", "\n ", "transforms: ", "[]", ",", "\n ", "validators: ", "[]", "]"])
    lib/conform/schema.ex:289: Conform.Schema.stringify/2
    lib/conform/schema.ex:268: Conform.Schema.write_quoted/2
    lib/mix/tasks/conform.new.ex:46: Mix.Tasks.Conform.New.do_run/2
    (mix) lib/mix/project.ex:196: Mix.Project.in_project/4
    (elixir) lib/file.ex:1137: File.cd!/2
    lib/mix/tasks/conform.new.ex:20: anonymous fn/3 in Mix.Tasks.Conform.New.run/1
    (elixir) lib/enum.ex:1387: Enum."-reduce/3-lists^foldl/2-0-"/3
```
1. [Not working as expected] After executing conform.configure, I checked the master_app.schema.exs, it's not yet picking up the port configs from app_one and app_two automatically. Sure enough, executing the master_app results in a crash from app_one not having a valid port.

**Comment by seantanly** [comment]
Also, is there a need for the master_app to include deps on exrm/conform? 

It looks like exrm/conform can run just fine from the umbrella root folder, and specifying deps on them on the root mix.exs seems sufficient.

**Comment by bitwalker** [comment]
Sorry for the delay, I've been quite busy unfortunately.

1. Unfortunately that's not how Mix configuration works out of the box. The only configuration taken into account is the top-level application. You can work around this via `import_config/1`, but the default does not merge in the configs of all dependencies.
2. Expected based on the above.
3. Noted, I'll get this addressed.
4. You need to add `extends: [:app_one, :app_two]` in your `master_app.schema.exs` file (after making sure they have schemas/confs. This will then implicitly use the schemas defined for those applications, while making them configurable via `master_app.conf`.
5. Yes, in order to run `mix release` commands or `mix conform.*` commands from a specific project, you have to have the dependencies added there.

**Comment by bitwalker** [comment]
The bug in `conform.new` has been fixed in `1.0.0-rc2` of conform.

**Comment by schnittchen** [comment]
When generating an umbrella app, the top level config.exs will contain

```
import_config "../apps/*/config/config.exs"
```
